src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb

```diff
@@ -352,11 +352,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 boundArguments.Add(boundArgument)
 
                 Dim elementType = GetTupleFieldType(boundArgument, argumentSyntax, diagnostics, hasErrors)
-                elementTypes.Add(elementType)
-
                 If elementType Is Nothing Then
                     hasNaturalType = False
+                    elementType = ReclassifyExpression(boundArgument, diagnostics).Type
```

Comment:
If that is what we have to do. We probably want to be very careful and have a special way to infer element types. ReclassifyExpression feels like to big of a hammer.